visualization/src/point_picking_event.cpp

```diff
@@ -68,8 +71,10 @@ pcl::visualization::PointPickingCallback::Execute (vtkObject *caller, unsigned l
       // Create a PointPickingEvent if a point was selected
       if (idx != -1)
       {
-        PointPickingEvent event (idx, x, y, z);
-        style->point_picking_signal_ (event);
+        CloudActorMapPtr cam_ptr = style->getCloudActorMap();
+        const auto actor = std::find_if(cam_ptr->cbegin(), cam_ptr->cend(), [this](const auto& cloud_actor) { return cloud_actor.second.actor.Get() == actor_; });
```

Comment:
```suggestion
        const auto actor = std::find_if(cam_ptr->cbegin(), cam_ptr->cend(), [this](const auto& cloud_actor) { return cloud_actor.second.actor.GetPointer () == actor_; });
```
This is likely the issue.

Reply:
actor_   values return  from vtk ,that not available in cam_ptr map second values so that return the empty value in cloud  name

Reply:
yes, its working